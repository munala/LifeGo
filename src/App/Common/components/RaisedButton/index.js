import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

import './styles.css';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
  },
});

const RaisedButton = ({
  label,
  classes,
  style,
  onClick,
  disabled,
}) => (
  <div className="container" >
    <Button
      className={classes.button}
      variant="contained"
      color="primary"
      onClick={onClick}
      disabled={disabled}
      style={style}
    >
      {label}
    </Button>
  </div>
);

RaisedButton.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  style: PropTypes.shape({}),
  classes: PropTypes.shape({}).isRequired,
};

RaisedButton.defaultProps = {
  disabled: false,
  style: undefined,
};

export default withStyles(styles)(RaisedButton);
