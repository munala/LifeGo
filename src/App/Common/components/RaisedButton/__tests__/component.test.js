import React from 'react';
import { shallow } from 'enzyme';
import RaisedButton from '../';

describe('RaisedButton tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<RaisedButton
      label="source"
      onClick={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
