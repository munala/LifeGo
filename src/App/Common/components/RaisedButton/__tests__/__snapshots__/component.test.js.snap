// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaisedButton tests renders without crashing 1`] = `
<RaisedButton
  classes={
    Object {
      "button": "RaisedButton-button-1",
    }
  }
  disabled={false}
  label="source"
  onClick={[Function]}
/>
`;
